jerry-core/parser/js/js-parser-expr.c

```diff
@@ -729,6 +726,42 @@ parser_parse_object_method (parser_context_t *context_p) /**< context */
 
   lexer_next_token (context_p);
 } /* parser_parse_object_method */
+
+/**
+ * Reparse the current literal as a common identifier.
+ *
+ * @return true - if the literal is reparsed
+ *         false - otherwise
+ */
+static bool
+parser_reparse_as_common_identifier (parser_context_t *context_p, /**< context */
+                                     parser_line_counter_t start_line, /**< start line */
+                                     parser_line_counter_t start_column) /**< start column */
+{
+  if (context_p->token.type != LEXER_RIGHT_BRACE
+      && context_p->token.type != LEXER_COMMA)
+  {
+    return false;
+  }
+
+  context_p->source_p = context_p->token.lit_location.char_p;
+  context_p->line = start_line;
+  context_p->column = start_column;
+
+  lexer_next_token (context_p);
+
+  if (context_p->token.type != LEXER_LITERAL
+      || context_p->token.lit_location.type != LEXER_IDENT_LITERAL)
+  {
+    parser_raise_error (context_p, PARSER_ERR_IDENTIFIER_EXPECTED);
+  }
+
+  lexer_construct_literal_object (context_p,
+                                  &context_p->token.lit_location,
+                                  context_p->token.lit_location.type);
```

Comment:
`context_p->token.lit_location.type` must be LEXER_IDENT_LITERAL, it is checked before.